### ISSUE TITLE
concurrent simplify?

### ISSUE THREAD

**Issue by ashander** [ashander/ftprime#56]
in principle the simplify step can be done on a separate thread

**Comment by ashander** [comment]
concern: seems a bit tricky to potentially avoid race conditions without expanding memory use.

**Comment by ashander** [comment]
@molpopgen commented

> In my case, I have 1,000 generations of table stored up on the C++ side, which makes it easy for me to swap them into a new container that msprime will act on.  

indicating we can probably do this easily after all code is using append columns (#41). seems like there is still a copy, but maybe nbd

**Comment by molpopgen** [comment]
> indicating we can probably do this easily after all code is using append columns (#41). seems like there is still a copy, but maybe nbd

Yes, in the case of using MPI, there is a copy via pickling moving to/from different Python processes.
